crates/store/src/db/models/queries/accounts.rs

```diff
@@ -565,6 +502,66 @@ pub(crate) fn select_account_storage_map_values(
     Ok(StorageMapValuesPage { last_block_included, values })
 }
 
+/// Returns account storage header at a given block by reading from `accounts.storage_header`
+/// and deserializing the storage header blob.
+pub(crate) fn select_account_storage_at_block(
+    conn: &mut SqliteConnection,
+    account_id: AccountId,
+    block_num: BlockNumber,
+) -> Result<AccountStorage, DatabaseError> {
```

Comment:
I'm not sure this method works correctly. Specifically, it is supposed to return the storage header, but seems to be returning fully account storage (which it tries to deserialize from the storage header).